src/api/core/organizations.rs

```diff
@@ -1484,3 +1532,324 @@ async fn import(org_id: String, data: JsonUpcase<OrgImportData>, headers: Header
 
     Ok(())
 }
+
+#[get("/organizations/<org_id>/groups")]
+async fn get_groups(org_id: String, _headers: AdminHeaders, conn: DbConn) -> JsonResult {
+    let groups = Group::find_by_organization(&org_id, &conn).await.iter().map(Group::to_json).collect::<Value>();
+
+    Ok(Json(json!({
+        "Data": groups,
+        "Object": "list",
+        "ContinuationToken": null,
+    })))
+}
+
+#[derive(Deserialize)]
+#[allow(non_snake_case)]
+struct GroupRequest {
+    Name: String,
+    AccessAll: Option<bool>,
+    ExternalId: String,
+    Collections: Vec<SelectionReadOnly>,
+}
+
+impl GroupRequest {
+    pub fn to_group(&self, organizations_uuid: &str) -> Result<Group, String> {
+        let access_all_value = match self.AccessAll {
+            Some(value) => value,
+            _ => return Err(String::from("Could not convert GroupRequest to Group, because AccessAll has no value!")),
```

Comment:
Prefer returning an `ApiResult<Group>` here:

```suggestion
            _ => err!("Could not convert GroupRequest to Group, because AccessAll has no value!"),
```